src/VTKBookLaTex/Chapter04.tex

```diff
@@ -64,102 +72,776 @@ \subsection{The Functional Model}
+
+\begin{figure}[!htb]
+  \centering
+  \includegraphics[width=0.8\textwidth]{Figure4-15}\\
+  \caption{Description of implicit execution process implemented in VTK. The Update() method is initiated via the Render() method from the actor. Data flows back to the mapper via the RequestData() method. Arrows connecting filter and data objects indicate direction of the Update() process.}\label{fig:Figure4-15}
+\end{figure}
+
+To understand the execution of the VTK pipeline, it is useful to view the process from several different vantage points. Note that each of the following figures is not completely accurate, rather they are presented as depictions whose purpose is to describe the important features of the process.
+
+Figure \ref{fig:Figure4-15} shows a simplified description of VTK's execution process. Generally the execution of the pipeline is triggered by a mapper's Render() method invocation, typically in response to a Render() method invocation on an associated vtkActor (which in turn receives it from the render window). Next, the Update() method is called on the input to the mapper (resulting in a cascade of method invocations requesting information and data). Eventually, data must be computed and returned to the object initiating the request, in this case the mapper. The RequestData() method actually executes the filter(s) in the pipeline and produces output data. Note the direction of flow --- here we define the direction of data flow as the \emph{downstream} direction, and the direction of the Update() invocation the \emph{upstream} direction.
+
+\begin{figure}[!htb]
+  \centering
+  \includegraphics[width=0.8\textwidth]{Figure4-16}\\
+  \caption{The logical relationship of the algorithm, executive and ports constituting a filter. The executive is responsible for managing the execution of the algorithm, and coordinating with information requests traveling through the pipeline. Ports correspond to logical, distinct inputs and outputs.}\label{fig:Figure4-16}
+\end{figure}
+
+The next figure, Figure \ref{fig:Figure4-16}, shows the relationship between the executive and the algorithm, which are paired to form a filter. This view of the filter is independent of the pipeline and contains all the information about the interface of the algorithm, namely the number and availability of inputs and outputs. Finally Figure \ref{fig:Figure4-17} shows the connections between filters. Notice that the output data object is not directly wired to the input connection. Rather the downstream filters's input connection is associated with the upstream filter's output port. This separation of data object from the input port means that data type checking can be deferred u
```

Comment:
I removed the reference to `LoopShrk.cxx` existing in the markdown version:
https://lorensen.github.io/VTKExamples/site/VTKBook/04Chapter4/#pipeline-execution-and-information-objects

The code references point to
https://lorensen.github.io/VTKExamples/site/VTKBookFigures/

And I've realized that many (virtually all?) of this files point to non-master branches/commits, such as
https://gitlab.kitware.com/vtk/vtk/blob/3ad5fa14aaa30c57cab400a7494bd6e2eac37bf1/graphics/examplesCxx/LoopShrk.cxx

Which do not exist in **VTK master**. That's why I removed it here.

I ignore whether this is intended. Please let me know if I am missing something, and if I should keep the reference to the `LoopShrk.cxx` file.